pkg/kubecost/asset.go

```diff
@@ -1044,16 +1044,17 @@ func (cm *ClusterManagement) String() string {
 
 // Disk represents an in-cluster disk Asset
 type Disk struct {
-	labels     AssetLabels
-	properties *AssetProperties
-	start      time.Time
-	end        time.Time
-	window     Window
-	adjustment float64
-	Cost       float64
-	ByteHours  float64
-	Local      float64
-	Breakdown  *Breakdown
+	labels       AssetLabels
+	properties   *AssetProperties
+	start        time.Time
+	end          time.Time
+	window       Window
+	adjustment   float64
+	Cost         float64
+	ByteHours    float64
+	Local        float64
+	Breakdown    *Breakdown
+	StorageClass string // @bingen:field[version=17]
```

Comment:
Will Kubernetes allow 2 disks of the same `AssetProperties` but of different storage classes? If it does, we may have to think deeper about how this might impact this https://github.com/opencost/opencost/blob/0094cba41fc0827e1598ceba9f6901739aed6386/pkg/kubecost/asset.go#L2744

Reply:
Kubernetes doesn't allow this. I can confirm that, ran the following test: 

I tried to create two storage classes `test-sc1 `and `test-sc2` and then created the same persistent volume name pv-common only changing the storage classes and i got the following error message 
`The PersistentVolumeClaim "pv-common" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests for bound claims
  core.PersistentVolumeClaimSpec{
  	... // 2 identical fields
  	Resources:        {Requests: {s"storage": {i: {...}, s: "10Gi", Format: "BinarySI"}}},
  	VolumeName:       "",
- 	StorageClassName: &"test-sc2",
+ 	StorageClassName: &"test-sc1",
  	VolumeMode:       &"Filesystem",
  	DataSource:       nil,
  	DataSourceRef:    nil,
  }`


Which only makes sense as Persistent volume is not namespace bound but persistent volume claim is namespace bound. i.e 2 different namespace can have same pvc name. There's also one to one mapping between PV and PVC. For two disk asset property to be same PV name should be same as it is part of the [asset property struct](https://github.com/avrodrigues5/opencost/blob/72c7d197a7d7ce3f9283312db979d8eab25496d0/pkg/kubecost/assetprops.go#L134).

Reply:
This is a good question. Let's check on that -- I'm guessing it's fine as-is, but we'll want to be sure.